services/oauth_service/authorization.go

```diff
@@ -0,0 +1,22 @@
+package oauth_service
+
+import (
+	"fmt"
+
+	"github.com/gutakk/go-google-scraper/models"
+
+	"github.com/go-oauth2/oauth2/v4/errors"
+)
+
+func PasswordAuthorizationHandler(username string, password string) (userID string, err error) {
+	user, findUserByErr := models.FindUserBy(&models.User{Email: username})
+	if findUserByErr != nil {
+		return "", errors.ErrInvalidClient
+	}
+
+	if err := models.ValidatePassword(user.Password, password); err != nil {
+		return "", errors.ErrInvalidClient
+	}
```

Comment:
@gutakk Regarding [our RFC](https://github.com/nimblehq/compass/issues/471) and [convention](https://twitter.com/SayCoolEnglish/status/1353289188041482241), we should reuse the error variable and also declare the error then check it later

```suggestion
	user, err := models.FindUserBy(&models.User{Email: username})
	if err != nil {
		return "", errors.ErrInvalidClient
	}

        err = models.ValidatePassword(user.Password, password)
	if err != nil {
		return "", errors.ErrInvalidClient
	}
```

Reply:
Fixed in 7024841be21a06fc896f90a0c8f09d722157fedd